flytekit/core/context_manager.py

```diff
@@ -325,6 +325,23 @@ def get(self, key: str) -> typing.Any:
         return self.__getattr__(attr_name=key)
 
 
+class _GroupSecrets(object):
+    """
+    This is a dummy class whose sole purpose is to support "attribute" style lookup for secrets
+    """
+
+    def __init__(self, group: str, sm: typing.Any):
+        self._group = group
+        self._sm = sm
+
+    def __getattr__(self, item: str) -> str:
+        """
+        Returns the secret that matches "group"."key"
+        the key, here is the item
+        """
+        return self._sm.get(self._group, item)
```

Comment:
can we make this an inner class to `SecretsManager`?

Reply:
can do